lambdas/handlers/delete_document_reference_handler.py

```diff
@@ -29,74 +26,36 @@
 )
 @override_error_check
 def lambda_handler(event, context):
+    logger.info("Delete Document Reference handler has been triggered")
+
+    nhs_number = extract_nhs_number_from_event(event)
+    doc_type = extract_document_type_as_enum(event["queryStringParameters"]["docType"])
+
     request_context.app_interaction = LoggingAppInteraction.DELETE_RECORD.value
-    nhs_number = event["queryStringParameters"]["patientId"]
-    doc_type = extract_document_type(event["queryStringParameters"]["docType"])
     request_context.patient_nhs_no = nhs_number
-    document_service = DocumentService()
+
+    deletion_service = DocumentDeletionService()
 
     try:
-        if doc_type == SupportedDocumentTypes.ALL.value:
-            arf_delete_response = handle_delete(
-                document_service, nhs_number, str(SupportedDocumentTypes.ARF.value)
-            )
-            lg_delete_response = handle_delete(
-                document_service, nhs_number, str(SupportedDocumentTypes.LG.value)
+        files_deleted = deletion_service.handle_delete(nhs_number, doc_type)
+        if files_deleted:
+            logger.info(
+                "Documents were deleted successfully", {"Result": "Successful deletion"}
             )
-
-            if (
-                arf_delete_response["statusCode"] == 404
-                and lg_delete_response["statusCode"] == 404
-            ):
-                return ApiGatewayResponse(
-                    404, "No documents available", "DELETE"
-                ).create_api_gateway_response()
-
             return ApiGatewayResponse(
                 200, "Success", "DELETE"
             ).create_api_gateway_response()
-
-        if doc_type == SupportedDocumentTypes.ARF.value:
-            return handle_delete(
-                document_service, nhs_number, str(SupportedDocumentTypes.ARF.value)
-            )
-
-        if doc_type == SupportedDocumentTypes.LG.value:
-            return handle_delete(
-                document_service, nhs_number, str(SupportedDocumentTypes.LG.value)
+        else:
+            logger.info(
+                "No records was found for given patient. No document deleted.",
+                {"Result": "No documents available"},
             )
+            return ApiGatewayResponse(
+                404, "No documents available", "DELETE"
+            ).create_api_gateway_response()
 
     except ClientError as e:
         logger.info(str(e), {"Result": f"Unsuccessful deletion due to {str(e)}"})
         return ApiGatewayResponse(
             500, "Failed to delete documents", "DELETE"
         ).create_api_gateway_response()
```

Comment:
Thanks :) 
I grabbed the LambdaException class from your branch and added a decorator to catch that. (and added unit tests around that) 
I am still keeping this ClientError except block as a safety net for AWS-related errors, but if any underlying code raises our custom LambdaException it will be handled now.